src/sim/sim_events.cc

```diff
@@ -54,19 +54,38 @@ namespace gem5
 
 GlobalSimLoopExitEvent::GlobalSimLoopExitEvent(Tick when,
                                                const std::string &_cause,
-                                               int c, Tick r)
+                                               int c, Tick r,
+                                               uint64_t hypercall_id,
+        std::map<std::string, std::string> payload)
```

Comment:
Why not create a new type for hypercalls that doesn't have the extraneous parameters. I don't think we'll ever need the code, repeat, or cause parameters for the hypercall. IMO, we should specialize more and reuse old code less.

In fact, I feel like this could be the other way around where we use the hypercall for the "old" way with the payload being `{'cause': <whatever>}`